### PR TITLE
Use the operator-registry image more consistently

### PR DESCRIPTION
**Pull request by dsimansk**
Changing the template to use the same base image on L1 and L20 consistently. 

Wrt/ https://github.com/openshift-knative/serverless-operator/pull/2495#issuecomment-1956319585

On a related subject PTAL on my comment in Jira, it seems to me that RHEL as a builder is unnecessary. But I might be missing some context. 
https://issues.redhat.com/browse/SRVCOM-2901?focusedId=24205814&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-24205814

/cc @mgencur @pierDipi

### CODE REVIEW

**Comment by pierDipi** [comment]
Thanks for looking into making CI more stable, I see there is a discussion on the issue, so I'll leave this to you and Martin to resolve.

**Comment by dsimansk** [comment]
@mgencur @pierDipi I was looking into simplifying the Dockerfile per my jira's comment. It might work with a bit of `USER root` magic. I'll do a new WIP PR to verify it. 

Anyway that doesn't block this PR, that polishes the template and uses same image for both layers it's being used.

**Comment by mgencur** [comment]
This PR looks good to me but there's a conflict.

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

File: olm-catalog/serverless-operator/index/Dockerfile
Comment (matzew) [inline olm-catalog/serverless-operator/index/Dockerfile:20]: :smile:  That's quiet a good fix.

Do you want that on 1.32 too, @dsimansk ?

Comment (ReToCode) [reply olm-catalog/serverless-operator/index/Dockerfile:20]: yes please ;)

Comment (dsimansk) [reply olm-catalog/serverless-operator/index/Dockerfile:20]: Sure, we can do that certainly.

Comment (dsimansk) [reply olm-catalog/serverless-operator/index/Dockerfile:20]: Neverming the cherry-pick will probably fail due to main being on 4.15 OCP_MAX variable. I'll defer to manual change then. :)

**Comment by dsimansk** [comment]
High deman for 4.14 clusters or smaller pool?
```
 WARN[2024-03-05T15:04:55Z] Found 1 conditions for ClusterClaim:
  *[2024-03-05T14:04:55Z]NoClusters: No clusters in pool are ready to be claimed 
INFO[2024-03-05T15:04:55Z] Releasing cluster claims for test test-upgrade-aws-414 
```

**Comment by dsimansk** [comment]
Shared cluster pools are getting kind of scars, seams so. 

Screenshot 2024-03-05 at 16 11 25

File: olm-catalog/serverless-operator/index/Dockerfile
Comment (ReToCode) [reply olm-catalog/serverless-operator/index/Dockerfile:20]: --> https://github.com/openshift-knative/serverless-operator/pull/2545

**Comment by dsimansk** [comment]
@ReToCode I'd override the check if it fails with unavailable cluster again. Given the upgrade test passed, and index image is buildable in all test instance apparently.

**Pull request merged**
